### PR TITLE
Token list service refac

### PR DESCRIPTION
**Pull request by brandonseydel**

### CODE REVIEW

**Review by dkent600 — CHANGES_REQUESTED** [review]
I know you didn't request my review on this.  But just in case, I just wanted to hold it up.  If you really want to submit it, i'll feedback.

**Comment by brandonseydel** [comment]
> I know you didn't request my review on this. But just in case, I just wanted to hold it up. If you really want to submit it, i'll feedback.

Yes please review.

File: src/services/services.ts [hunk 1]
```diff
@@ -30,17 +30,16 @@ export class Services {
...
-     * throws an error if targetNetwork is no good
-     */
-    await this.ethereumService.initialize(targetNetwork);
-    await this.contractsDeploymentService.initialize(targetNetwork);
     this.contractsService.initialize();
     this.ipfsService.initialize(this.kolektivoService);
-    await this.tokenService.initialize();
+    return Promise.all([
+      this.ethereumService.initialize(targetNetwork),
+      this.contractsDeploymentService.initialize(targetNetwork),
```
Comment (dkent600) [inline src/services/services.ts:40]: ethereumService and contractsDeploymentService must be initialized before contractsService

File: src/services/token-list-service.ts [hunk 2]
```diff
@@ -22,91 +23,69 @@ export class TokenListService {
...
-      this.timingService.endTimer('fetch tokeninfos');
-      this.tokenInfos = tokenInfoMap;
+    if (this.tokenInfos) return;
+
+    this.timingService.startTimer('fetch tokeninfos');
+    const uris = this.ethereumService.targetedNetwork && TokenLists.get(this.ethereumService.targetedNetwork);
+
+    if (!uris) {
+      this.logger.fatal(`TokenListProvider: no tokenlists available for the network ${this.ethereumService.targetedNetwork ?? ''}`);
+      return;
```
Comment (dkent600) [inline src/services/token-list-service.ts:35]: should end the timer before returning.

In general I am in favor of returning from functions only from a single location.

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/app.ts [hunk 3]
```diff
@@ -44,8 +44,10 @@ export class App {
     this.platform.window.removeEventListener('resize', this.recalc);
   }
 
-  binding(): Promise<void> {
-    return this.store.initializeServices();
+  binding() {
+    return this.store
+      .initializeServices()
+      .catch(() => void this.notificationService.toast({ type: 'danger', message: 'There was an error initializing the application' }));
```
Comment (dkent600) [inline src/app.ts:50]: Should also log the details (error message and stack trace)

**Review by dkent600 — CHANGES_REQUESTED** [review]

**Review by dkent600 — APPROVED** [review]

**Pull request merged**